vm/map.c

```diff
@@ -1052,30 +1052,36 @@ void vm_mapGetStats(size_t *allocsz)
 }
 
 
-vm_map_t *vm_getSharedMap(syspage_program_t *prog, int map)
+vm_map_t *vm_getSharedMap(syspage_prog_t *prog, int map)
 {
 #ifdef NOMMU
-	if (map < 0)
-		map = prog->dmap;
+	unsigned int i;
 
-	if (map >= 0 && map < 16)
-		return map_common.maps[map];
+	if (map < 0)
+		map = prog->dmaps[0];
 
+	for (i = 0; i < prog->dmapSz; ++i) {
+		if (map == prog->dmaps[i])
+			return map_common.maps[i];
+	}
 #endif
+
 	return NULL;
 }
 
 
+#ifdef NOMMU
```

Comment:
Move this `ifdef` to the inside of the function, make it return zero in `else` case

Reply:
Done